### PR TITLE
4616 checkbox

### PR DESCRIPTION
**Pull request by AdrianRosAdapt**

### CODE REVIEW

**Comment by AdrianRosAdapt** [comment]
Please don't merge. Created for reference purpose.

File: components/src/checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  }
+  if (checkboxType === "Warning" && isDarkTheme) {
+    accentColor = "accent-warning-dark"
+  } else if (checkboxType === "Warning" && !isDarkTheme) {
+    accentColor = "accent-warning-light"
+  }
+
+  const disabled = isDisabled ? true : false
+  return (
+    <input
```
Comment (snony14) [inline components/src/checkbox/Checkbox.tsx:54]: you may want to use the shorthand notation `<input {...props} />`

File: components/src/checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  isDarkTheme: boolean
+}
+export const Checkbox = ({
+  size,
+  checkboxType,
+  isDisabled,
+  text,
+  isDarkTheme,
+}: CheckboxProps) => {
+  let base, accentColor
```
Comment (snony14) [inline components/src/checkbox/Checkbox.tsx:19]: You can potentially define this as follow` const base = useMemo(()=>{if (size === "small") {
    return "w-3 h-3"
  } else if (size === "medium") {
    return "w-4 h-4"
  }},[size])`

File: components/src/checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  if (size === "small") {
+    base = "w-3 h-3"
+  } else if (size === "medium") {
+    base = "w-4 h-4"
+  }
+  if (checkboxType === "Primary" && isDarkTheme) {
+    accentColor = "accent-primary-dark"
```
Comment (snony14) [inline components/src/checkbox/Checkbox.tsx:26]: same can go with `accentColor`, define it as a const

File: components/src/checkbox/Checkbox.tsx (hunk 1 above)
Comment (AdrianRosAdapt) [reply components/src/checkbox/Checkbox.tsx:54]: I am not sure what you mean here to be honest.

File: components/src/checkbox/Checkbox.tsx (hunk 2 above)
Comment (AdrianRosAdapt) [reply components/src/checkbox/Checkbox.tsx:19]: Would the following code be better ? `  const base = size === "small" ? "w-3 h-3" : "w-4 h-4"`

File: components/src/checkbox/Checkbox.tsx (hunk 1 above)
Comment (AdrianRosAdapt) [reply components/src/checkbox/Checkbox.tsx:54]: Oh I see, I get it now. Thank you

**Comment by algreasley** [comment]
> Please don't merge. Created for reference purpose.

==> can use "Draft" if you would like comments but not open for formal review yet ==>

**Comment by AdrianRosAdapt** [comment]
> > Please don't merge. Created for reference purpose.
> 
> ==> you can use "Still in progress? Convert to Draft" in the top right if you would like comments but not open for formal review yet ==>

I did not know that. Thank you for that. It is ready for a review now tho. Not sure if I should remove the comment or not.

**Review by adamshcao — APPROVED** [review]

**Pull request merged**